test/qa/test_routing.py

```diff
@@ -27,86 +29,145 @@ def setup_function(function):
 def teardown_function(function):
     logging.log(data=info.collect())
     logging.log()
-    lib.set_routing("on")
 
 
+TABLE = 205
+SUBNET_1 = "1.1.1.1"
+SUBNET_2 = "2.2.2.2"
+SUBNET_3 = "3.3.3.3"
+
+
+@pytest.mark.parametrize("tech,proto,obfuscated", lib.TECHNOLOGIES)
 @pytest.mark.flaky(reruns=2, reruns_delay=90)
 @timeout_decorator.timeout(40)
-def test_routing_on():
-    subnet1 = "1.1.1.1"
-    subnet2 = "2.2.2.2"
-    subnet3 = "3.3.3.3"
-    lib.add_subnet_to_allowlist(f"{subnet1}/32")
-    lib.add_subnet_to_allowlist(f"{subnet2}/32")
-    lib.add_subnet_to_allowlist(f"{subnet3}/32")
-    table = 205
-
-    with lib.ErrorDefer(sh.nordvpn.allowlist.remove.all):
-        with lib.ErrorDefer(sh.nordvpn.disconnect):
-            output = sh.nordvpn.connect()
-            assert lib.is_connect_successful(output)
-
-            rules = sh.ip.rule.show.table(table)
-            assert "fwmark" in rules
-            policyRoutes = sh.ip.route.show.table(table)
-            assert subnet1 in policyRoutes
-            assert subnet2 in policyRoutes
-            assert subnet3 in policyRoutes
-            assert "nordlynx" in policyRoutes
+def test_routing_enabled_connect(tech, proto, obfuscated):
+    lib.set_technology_and_protocol(tech, proto, obfuscated)
+
+    lib.add_subnet_to_allowlist(f"{SUBNET_1}/32")
+    lib.add_subnet_to_allowlist(f"{SUBNET_2}/32")
+    lib.add_subnet_to_allowlist(f"{SUBNET_3}/32")
+
+    with lib.Defer(sh.nordvpn.allowlist.remove.all):
+        with lib.Defer(sh.nordvpn.disconnect):
+            print(sh.nordvpn.connect())
+            assert network.is_available()
 
+            assert "fwmark" in sh.ip.rule.show.table(TABLE)
+            policy_routes = sh.ip.route.show.table(TABLE)
+            assert SUBNET_1 in policy_routes
+            assert SUBNET_2 in policy_routes
+            assert SUBNET_3 in policy_routes
 
+            network_interface = "nordtun" if tech == "openvpn" else "nordlynx"
+            assert network_interface in policy_routes
+
+            assert settings.get_is_routing_enabled()
+
+
+@pytest.mark.skip("LVPN-3273; LVPN-1574")
```

Comment:
Could you add a comment to the tickets to enable the test when fixed?

Reply:
Good point :smile: